gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -220,14 +255,27 @@ public ApiCallContext merge(ApiCallContext inputCallContext) {
       newChannelAffinity = this.channelAffinity;
     }
 
+    ImmutableListMultimap.Builder<String, String> newExtraHeadersBuilder =
+        ImmutableListMultimap.builder();
+    if (grpcCallContext.extraHeaders != null) {
```

Comment:
Please consider ensuring (in outer code) that `extraHeaders` is always non-null, then remove this check (if it fails with nullpointer it would be a fail-fast behavior allowing to catch bugs in out code quickly).

Reply:
In this case remove `@Nullable` for `extraHeaders`?

Reply:
Yes. Actually it is a private method, using `@Nullable` in private methods is a bit too much (ok, if you wish to, but definitely not a requirement).